torchvision/transforms/functional_tensor.py

```diff
@@ -430,16 +430,13 @@ def resize(
     img: Tensor,
     size: List[int],
     interpolation: str = "bilinear",
-    antialias: Optional[bool] = None,
+    antialias: bool = False,
 ) -> Tensor:
     _assert_image_tensor(img)
 
     if isinstance(size, tuple):
         size = list(size)
 
-    if antialias is None:
-        antialias = False
-
```

Comment:
I think this is a BC-breaking change for users who passed `antialias=None`.

Considering the relatively low benefit of removing support for `None`, we might want to keep it to avoid any potential issues?

Reply:
This function is importable without a leading underscore as

```py
from torchvision.transforms.functional_tensor import resize
```

and as a result, it is likely that most users will consider it public, as this is a widely used convention throughout the pydata ecosystem.

I think it's fine to modify the prototype, but I'd rather keep the stable area untouched to avoid any potential breakage. Especially considering the low benefit, and the fact that these transforms will eventually be superseded by the prototype ones.

Reply:
Yeah, we should have added leading underscore to them. It very unfortunate if people today rely on this undocumented API. In this PR https://github.com/pytorch/vision/pull/3200 we removed "PRIVATE METHOD" in the docstring.
Previously I already refactored and removed an arg from `resize` method in `functional_pil` and `functional_tensor` https://github.com/pytorch/vision/pull/6191 . 
Should we restore the arg for BC and warn about new behaviour ?

Reply:
Sorry just realized this is a public function, thanks @NicolasHug for noticing.
I agree that we shouldn't make BC changes in this case.

Looking at the update of @vfdev-5 comment, I think `resize_image_tensor` is under prototype which should be okay to change explicitly, but lets keep functional_tensor.resize unchanged.

Reply:
> Yeah, we should have added leading underscore to them. It very unfortunate if people today rely on this undocumented API

I agree. Unfortunately right now, there is literally nothing that distinguishes `torchvision.transforms.functional` from `torchvision.transforms.functional_tensor`. So we can't assume users know that one is private while the other one isn't.

> Previously I already refactored and removed an arg from resize method in functional_pil and functional_tensor https://github.com/pytorch/vision/pull/6191 .
Should we restore the arg for BC and warn about new behaviour ?


Yes, I think this is safer. What do you mean about "warning about new behaviour" though? We can't really do silent changes either in this case, we would have to go through deprecations.

Reply:
> Unfortunately right now, there is literally nothing that distinguishes torchvision.transforms.functional from torchvision.transforms.functional_tensor

The only diff is that public methods are documented and visible in the docs. Private methods do not have any docstring and invisible in the docs. 

> What do you mean about "warning about new behaviour" though? We can't really do silent changes either in this case, we would have to go through deprecations.

Yes, for public API is it the right way to proceed. I still have them in private space that's why was suggesting something in between.

I think we should decide and make things more clear with `functional_tensor` and `functional_pil`. If they are private, let's put them to `_functional_tensor.py` and `_functional_pil.py` with all necessary warnings/deprecations etc.

Reply:
> I still have them in private space

Sorry what do you mean? We should treat everything in `functional_tensor` and `functional_pil` as public

Reply:
> After that the decision was made to remove the documentation warning

Decision was very quick: https://github.com/pytorch/vision/pull/3069#pullrequestreview-541808920 and https://github.com/pytorch/vision/pull/3200